fv_be/app/admin/base_admin.py

```diff
@@ -70,16 +71,30 @@ def item_id(self, instance):
     item_id.short_description = _("Id")
 
     def admin_link(self, instance):
-        url = reverse(
-            f"admin:{instance._meta.app_label}_{instance._meta.model_name}_change",
-            args=(instance.id,),
-        )
-        # todo: i18n for 'Edit' not working here for some reason
-        return format_html('<a href="{}">{}: {}</a>', url, "Edit", str(instance))
+        # todo: Add reverse links for all models
+        try:
+            url = reverse(
+                f"admin:{instance._meta.app_label}_{instance._meta.model_name}_change",
+                args=(instance.id,),
+            )
+            # todo: i18n for 'Edit' not working here for some reason
+            return format_html('<a href="{}">{}: {}</a>', url, "Edit", str(instance))
+        except NoReverseMatch as e:
+            return None
 
     def save_model(self, request, obj, form, change):
         if not change:
             obj.created_by = request.user
 
         obj.last_modified_by = request.user
         super().save_model(request, obj, form, change)
+
+
+# todo: better name for following class
+class AdminHideUtility(BaseAdmin):
+    """
+    This utility class lets the admin classes to be registered on the top level of admin interface, to get the reverse
+    links working for inline admin classes and also prevents the classes that inherit this to be shown at the top level.
+    """
```

Comment:
This is a bit of a surprising implementation-- can you please add a comment with a link to a url that explains how this works?

Reply:
Yes, I could not find the official documentation for the has_model_perms method. From the [stackoverflow link](https://stackoverflow.com/questions/2431727/django-admin-hide-a-model), it seems it may have been used in the previous version of Django ?
I've updated it with another methods which I believe does the same thing and has been documentation for the current Django version.